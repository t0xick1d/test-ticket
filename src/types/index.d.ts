// declare module '*.module.scss' {
//    const classes: { [key: string]: string };
//    const content: React.FunctionComponent<React.SVGAttributes<SVGElement>>;
//    export default classes;
// }
