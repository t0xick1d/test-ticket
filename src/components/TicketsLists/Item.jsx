import React from 'react';
import style from './style.module.scss';

const Item = ({ data }) => {
   console.log(data);

   const formatMinutesToHours = (minutes) => {
      const hours = Math.floor(minutes / 60);
      const remainingMinutes = minutes % 60;

      return `${hours}ч ${remainingMinutes}м`;
   };
   return (
      <li className={style.containerItem}>
         <div className={style.containerMainInf}>
            <p className={style.priceText}>{data.price} P</p>
            <p className={style.carrierText}>{data.carrier}</p>
         </div>
         <ul className={style.containerSegments}>
            {data.segments.map((e) => {
               return (
                  <li className={style.containerSegmentsItem}>
                     <div style={{ width: 120 }}>
                        <p style={{ color: '#adbbc3', fontSize: 12, textTransform: 'uppercase' }}>
                           MOW - HKT
                        </p>
                        <p>{e.date}</p>
                     </div>
                     <div style={{ width: 120 }}>
                        <p style={{ color: '#adbbc3', fontSize: 12, textTransform: 'uppercase' }}>
                           В пути
                        </p>
                        <p>{formatMinutesToHours(e.duration)}</p>
                     </div>
                     <div style={{ width: 120 }}>
                        <p style={{ color: '#adbbc3', fontSize: 12, textTransform: 'uppercase' }}>
                           {e.stops.length} пересадки
                        </p>
                        <div style={{ display: 'flex', justifyContent: 'center' }}>
                           {e.stops.map((e) => (
                              <p style={{ padding: 2, alignItems: 'center' }}>{e}</p>
                           ))}
                        </div>
                     </div>
                  </li>
               );
            })}
         </ul>
      </li>
   );
};

export default Item;
