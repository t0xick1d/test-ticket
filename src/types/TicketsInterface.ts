export interface TicketsI {
   price: number;
   carrier: number;
   date: string;
   segments: [SegmentsI];
}
export interface SegmentsI {
   origin: string;
   destination: string;
   departureTime: string;
   duration: number;
   stops: string[];
}
