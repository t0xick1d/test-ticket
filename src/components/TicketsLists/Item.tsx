import React from 'react';
import style from './style.module.scss';
import { TicketsI } from '../../types/TicketsInterface';

interface MyComponentProps {
   data: TicketsI;
}

const Item: React.FC<MyComponentProps> = ({ data }) => {
   const formatMinutesToHours = (minutes: number) =>
      `${Math.floor(minutes / 60)}ч ${minutes % 60}м`;
   const getHoursAndMinutes = (datetimeStr: string) =>
      `${new Date(datetimeStr).getHours().toString().padStart(2, '0')}:${new Date(datetimeStr)
         .getMinutes()
         .toString()
         .padStart(2, '0')}`;

   return (
      <>
         <div className={style.containerMainInf}>
            <p className={style.priceText}>{data.price} P</p>
            <p className={style.carrierText}>{data.carrier}</p>
         </div>
         <ul className={style.containerSegments}>
            {data.segments.map((e, i) => (
               <li key={`segment${i}`} className={style.containerSegmentsItem}>
                  <div className={style.containerInfo}>
                     <p className={style.secondaryText}>MOW - HKT</p>
                     <p>{getHoursAndMinutes(e.departureTime)}</p>
                  </div>
                  <div className={style.containerInfo}>
                     <p className={style.secondaryText}>В пути</p>
                     <p>{formatMinutesToHours(e.duration)}</p>
                  </div>
                  <div className={style.containerInfo}>
                     <p className={style.secondaryText}>{e.stops.length} пересадки</p>
                     <div className={style.containerStops}>
                        {e.stops.map((e, i) => (
                           <p key={i} className={style.pStops}>
                              {e}
                           </p>
                        ))}
                     </div>
                  </div>
               </li>
            ))}
         </ul>
      </>
   );
};

export default Item;
