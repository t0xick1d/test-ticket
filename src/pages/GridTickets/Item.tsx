import React from 'react';
import { TicketsI } from 'types/TicketsInterface';
import { formatMinutesToHours, getHoursAndMinutes } from '../../utils/timeUtils/timeUtils';

import style from './style.module.scss';

const Item: React.FC<{ row: TicketsI }> = ({ row }) => {
   return (
      <div className={style.containerSegments}>
         {row.segments.map((e, i) => {
            return (
               <li key={`segment${i}`} className={style.containerSegmentsItem}>
                  <div className={style.containerInfo}>
                     <p className={style.secondaryText}>MOW - HKT</p>
                     <p>{getHoursAndMinutes(e.departureTime)}</p>
                  </div>
                  <div className={style.containerInfo}>
                     <p className={style.secondaryText}>В пути</p>
                     <p>{formatMinutesToHours(e.duration)}</p>
                  </div>
                  <div className={style.containerInfo}>
                     <p className={style.secondaryText}>{e.stops.length} пересадки</p>
                     <div className={style.containerStops}>
                        {e.stops.map((e, i) => (
                           <p key={i} className={style.pStops}>
                              {e}
                           </p>
                        ))}
                     </div>
                  </div>
               </li>
            );
         })}
      </div>
   );
};

export default Item;
