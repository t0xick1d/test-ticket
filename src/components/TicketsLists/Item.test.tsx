import React from 'react';
import { render, screen } from '@testing-library/react';
import Item from './Item';
import { TicketsI } from '../../types/TicketsInterface';

describe('Item component', () => {
   const mockTicket: TicketsI = {
      price: 13196,
      carrier: 'S7',
      date: '2025-06-01',
      segments: [
         {
            origin: 'MOW',
            destination: 'HKT',
            departureTime: '2025-06-17T10:30:00',
            duration: 120,
            stops: ['DXB', 'IST'],
         },
      ],
   };

   it('renders price and carrier correctly', () => {
      render(<Item data={mockTicket} />);
      expect(screen.getByText(/13196 P/)).toBeInTheDocument();
      expect(screen.getByText(/S7/)).toBeInTheDocument();
   });

   it('renders formatted departure time and duration', () => {
      render(<Item data={mockTicket} />);
      expect(screen.getByText('10:30')).toBeInTheDocument();
      expect(screen.getByText('2ч 0м')).toBeInTheDocument();
   });

   it('renders number of stops and each stop', () => {
      render(<Item data={mockTicket} />);
      expect(screen.getByText(/2 пересадки/)).toBeInTheDocument();
      expect(screen.getByText('DXB')).toBeInTheDocument();
      expect(screen.getByText('IST')).toBeInTheDocument();
   });

   it('matches snapshot', () => {
      const { asFragment } = render(<Item data={mockTicket} />);
      expect(asFragment()).toMatchSnapshot();
   });
});
