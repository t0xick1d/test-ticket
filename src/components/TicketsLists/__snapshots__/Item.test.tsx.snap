// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="containerMainInf"
  >
    <p
      class="priceText"
    >
      13196 P
    </p>
    <p
      class="carrierText"
    >
      S7
    </p>
  </div>
  <ul
    class="containerSegments"
  >
    <li
      class="containerSegmentsItem"
    >
      <div
        class="containerInfo"
      >
        <p
          class="secondaryText"
        >
          MOW - HKT
        </p>
        <p>
          10:30
        </p>
      </div>
      <div
        class="containerInfo"
      >
        <p
          class="secondaryText"
        >
          В пути
        </p>
        <p>
          2ч 0м
        </p>
      </div>
      <div
        class="containerInfo"
      >
        <p
          class="secondaryText"
        >
          2 пересадки
        </p>
        <div
          class="containerStops"
        >
          <p
            class="pStops"
          >
            DXB
          </p>
          <p
            class="pStops"
          >
            IST
          </p>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
